src/dbnode/persist/fs/files.go

```diff
@@ -1220,3 +1348,77 @@ func snapshotIndexSegmentFilePathFromTimeAndIndex(
 	suffix := filesetIndexSegmentFileSuffixFromTime(t, segmentIndex, segmentFileType)
 	return filesetPathFromTimeAndIndex(prefix, t, snapshotIndex, suffix)
 }
+
+func snapshotMetadataFilePathFromIdentifier(prefix string, id SnapshotMetadataIdentifier) string {
+	return path.Join(
+		prefix,
+		snapshotDirName,
+		fmt.Sprintf(
+			"%s%s%s%s%d%s%s%s",
+			snapshotFilePrefix, separator,
+			sanitizeUUID(id.UUID.String()), separator,
+			id.Index, separator,
+			metadataFileSuffix, fileSuffix))
+}
+
+func snapshotMetadataCheckpointFilePathFromIdentifier(prefix string, id SnapshotMetadataIdentifier) string {
+	return path.Join(
+		prefix,
+		snapshotDirName,
+		fmt.Sprintf(
+			"%s%s%s%s%d%s%s%s%s%s",
+			snapshotFilePrefix, separator,
+			sanitizeUUID(id.UUID.String()), separator,
+			id.Index, separator,
+			metadataFileSuffix, separator,
+			checkpointFileSuffix, fileSuffix))
+}
+
+// sanitizeUUID strips all instances of separator ("-") in the provided UUID string. This prevents us from
+// treating every "piece" of the UUID as a separate fragment of the name when we split filepaths by
+// separator. This works because the UUID library can still parse stripped UUID strings.
+func sanitizeUUID(uuid string) string {
```

Comment:
Can you make this function perhaps take a native UUID value, i.e. `func sanitizeUUIDString(value uuid.UUID) string` so it's just a little more type safe?